src/GitHub.VisualStudio/Services/UsageTracker.cs

```diff
@@ -1,319 +1,105 @@
         {
             // The services needed by the usage tracker are loaded when they are first needed to
             // improve the startup time of the extension.
-            if (userSettings == null)
+            if (!initialized)
             {
                 await ThreadingHelper.SwitchToMainThreadAsync();
 
                 client = gitHubServiceProvider.TryGetService<IMetricsService>();
                 connectionManager = gitHubServiceProvider.GetService<IConnectionManager>();
                 userSettings = gitHubServiceProvider.GetService<IPackageSettings>();
                 vsservices = gitHubServiceProvider.GetService<IVSServices>();
-
-                var program = gitHubServiceProvider.GetService<IProgram>();
-                storePath = System.IO.Path.Combine(
-                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
-                    program.ApplicationName,
-                    StoreFileName);
+                initialized = true;
             }
         }
 
-        async Task<UsageStore> LoadUsage()
+        async Task IncrementLaunchCount()
         {
-            await Initialize();
+            var usage = await LoadUsage();
+            ++usage.Model.NumberOfStartups;
+            ++usage.Model.NumberOfStartupsWeek;
+            ++usage.Model.NumberOfStartupsMonth;
+            await service.WriteLocalData(usage);
+        }
 
-            var json = fileExists(storePath) ? readAllText(storePath, Encoding.UTF8) : null;
-            UsageStore result = null;
-            try
-            {
-                result = json != null ?
-                    SimpleJson.DeserializeObject<UsageStore>(json) :
-                    new UsageStore { Model = new UsageModel() };
-            }
-            catch
-            {
-                result = new UsageStore { Model = new UsageModel() };
-            }
+        async Task<UsageData> LoadUsage()
+        {
+            await Initialize();
 
+            var result = await service.ReadLocalData();
             result.Model.Lang = CultureInfo.InstalledUICulture.IetfLanguageTag;
             result.Model.AppVersion = AssemblyVersionInformation.Version;
             result.Model.VSVersion = vsservices.VSVersion;
-
             return result;
         }
 
-        void SaveUsage(UsageStore store)
-        {
-            dirCreate(System.IO.Path.GetDirectoryName(storePath));
-            var json = SimpleJson.SerializeObject(store);
-            writeAllText(storePath, json, Encoding.UTF8);
-        }
-
-        void TimerTick(object state)
-        {
-            TimerTick()
-                .Catch(ex =>
-                {
-                    //log.Warn("Failed submitting usage data", ex);
-                })
-                .Forget();
-        }
-
         async Task TimerTick()
         {
             await Initialize();
 
-            if (firstRun)
+            if (firstTick)
             {
                 await IncrementLaunchCount();
```

Comment:
`IncrementLaunchCount` calls `LoadUsage` which calls `await Initialize` which has already been called right before this. Do we need these double `await Initialize` calls?

Reply:
Actually, yes we can remove the `Initialize` at the top of `TimerTick`! 👍